sdks/java/build-tools/beam-linkage-check.sh

```diff
@@ -84,9 +84,9 @@ function runLinkageCheck () {
 BRANCH_COMMIT=`git rev-parse --short=8 HEAD`
 runLinkageCheck $BRANCH_COMMIT
 
-git checkout master
-git pull
-MASTER_COMMIT=`git rev-parse --short=8 HEAD`
+git fetch
+MASTER_COMMIT=`git rev-parse --short=8 master`
```

Comment:
```suggestion
MASTER_COMMIT=`git rev-parse --short=8 origin/master`
```

local master and remote master branch are different until “git pull”

Reply:
Does line 71 `local BRANCH_NAME=$(git symbolic-ref --short HEAD) ` work?

Reply:
Line 71 would fail:

```
suztomo@suxtomo24:~/cloud-opensource-java$ MASTER_COMMIT=`git rev-parse --short=8 origin/master`
suztomo@suxtomo24:~/cloud-opensource-java$ git -c advice.detachedHead=false checkout $MASTER_COMMIT
HEAD is now at eebf0ce4 update http client (#1220)
suztomo@suxtomo24:~/cloud-opensource-java$ git symbolic-ref --short HEAD
fatal: ref HEAD is not a symbolic ref
suztomo@suxtomo24:~/cloud-opensource-java$ echo $?
128
```

Reply:
oh yes sorry shame on me, let me take a look, fix it and test it (never consider anything so obvious to not test it sorry).